### ISSUE TITLE
Averager testing

### ISSUE THREAD

**Issue by bennahugo** [ska-sa/xova#8]
@sjperkins @sphemakh @o-smirnov I'm reasonably happy this works as advertised. I've Performed basic visibility inspection after averaging with XOVA to test the averager in production use - as a drop in replacement in my polarization calibration pipeline. 
From #7, after adding a switch to average "CORRECTED_DATA" I can confirm that the flags are correctly handled - the percentages are spot on. I also run through a basic pol cal to make sure that the data can be calibrated succesfully. Indeed there are no problems here:

Spectrum is spot on:
![image](https://user-images.githubusercontent.com/7934586/73191829-ceae8180-4130-11ea-838d-64e1821f21e8.png)
Phase is spot on:
![image](https://user-images.githubusercontent.com/7934586/73191926-f998d580-4130-11ea-9d12-1a13446ec11d.png)

# Prepolcal:
### PKSB1934-638:
![image](https://user-images.githubusercontent.com/7934586/73190651-fbfa3000-412e-11ea-8f8d-e4e7c5c080ca.png)
![image](https://user-images.githubusercontent.com/7934586/73191002-7c209580-412f-11ea-81c4-e3f2efb3eb10.png)
### 3C286 (coloured per correlation - crosshands at about 10% linearly polarized):

Original dataset:
![image](https://user-images.githubusercontent.com/7934586/73193455-441b5180-4133-11ea-962b-26b801673d44.png)
![image](https://user-images.githubusercontent.com/7934586/73194049-39ad8780-4134-11ea-9a49-2bba7e1420ca.png)

Average dataset:
![image](https://user-images.githubusercontent.com/7934586/73192191-58f6e580-4131-11ea-914c-43f15b0ea6a0.png)
![image](https://user-images.githubusercontent.com/7934586/73192489-c571e480-4131-11ea-94cf-5032616931ac.png)

# Postpolcal (Ignore RFI affected bands - cannot be leakage calibrated):
### PKSB1934-638
![image](https://user-images.githubusercontent.com/7934586/73191799-c22a2900-4130-11ea-8a1e-ac433f7f8e96.png)
### 3C286:
![image](https://user-images.githubusercontent.com/7934586/73192264-71ff9680-4131-11ea-999f-0c8dd248b1b1.png)
![image](https://user-images.githubusercontent.com/7934586/73192354-95c2dc80-4131-11ea-896d-cd8afaccdc72.png)

**Comment by bennahugo** [comment]
It is at least an order of magnitude faster to average this compared to oldsplit

**Comment by bennahugo** [comment]
ok just checked the resulting images to ensure the uv coordinates are correct. Everything seems to be in place the maximum error is about e-4 down from the peak

**Comment by sjperkins** [comment]
@bennahugo the UVW coordinates are lerped, I guess if that ever becomes a problem then spline fitting may be appropriate.

**Comment by bennahugo** [comment]
I don't think it is a problem @sjperkins everything seems to be well within smearing tolerances

**Comment by sphemakh** [comment]
Great stuff!

**Comment by bennahugo** [comment]
@sjperkins if you give me permission to push I can push my small fixes up
